Sources/core/objects/Movable.h

```diff
@@ -34,7 +34,7 @@ class IMovable : public IOnMapObject
     void processMove();//for move
     virtual bool TryMove(Dir direct);
     bool CheckMoveTime();
-    bool CheckPassable();
+    virtual bool CheckPassable();
     bool MainMove();
```

Comment:
Yeah I was thinking that the bullets speed decrease with the time but I think that in this case a new function that manages the speed change should handle the forces too, and it could return a false when the speed is 0 so the process will delete it.